STYLEGUIDE.md

```diff
@@ -0,0 +1,83 @@
+
+Thanks for contributing to Lem! Here are some guidelines that we follow in the codebase.
+
+In general, we follow the Google guide: https://google.github.io/styleguide/lispguide.xml
+
+
+## Alexandria and other util libraries
+
+Lem depends on `alexandria`, so you can use `if-let`, `when-let` and similar functions.
+
+You can `:import-from` these symbols in your package definition, use the `alexandria:` package prefix, and define a package-local nickname.
+
+Try to not use new utility functions that you don't see in the codebase yet.
+
+For instance, we won't use `alexandria-2:line-up-first`.
+
+Try to not use `alexandria:curry` and prefer higher-order functions.
+
+
+## Deprecation warnings
+
+If you change or delete a feature, if only a variable name, you must
+take care of deprecation warnings. A user should be notified that
+something changed. If possible, her old config file should not fail
+loading, or it should not fail without notice. Measures can vary. Ideas:
+
+- don't delete or rename a `defvar`, a `defparameter` or a function, but leave it and if it is used, signal a warning or an error. Add a `;; DEPRECATED` comment with the date of the comment.
+- document the breaking change on the "next release changelog" issue (like https://github.com/lem-project/lem/issues/1027 or equivalent).
+
+
+## Documentation
+
+Write thorough docstrings to interactive commands (`define-command`),
+give meaningful documentation to important functions, give a
+high-level overview in packages and comments (the "why", not the
+"how"). Use the `:documentation` option of packages, generic functions
+and CLOS slots.
+
+Please also contribute to Lem's website if you add or change a feature.
+
+https://github.com/lem-project/lem-project.github.io/
+
+or at least, open an issue about it so we don't forget to do it, thank you.
+
+
+## Errors in Lem
+
+Don't signal an `error` but an `editor-error`. This one is displayed nicely to the user.
```

Comment:
This depends on the situation.
In some cases, it is better to use error than editor-error in places where an internal error should be used.